### ISSUE TITLE
Limited joining collector

### ISSUE THREAD

**Issue by amaembo** [amaembo/streamex#16]
Implement collector which joins stream of CharSequences with given delimiter which may short-circuit when the given length is reached adding the optional ellipsis. Possible features and settings include:

* [x] delimiter CharSequence
* [x] prefix/suffix CharSequences (to fully supercede the JDK Collectors.joining)
* [x] ellipsis CharSequence (default: "...")
* [x] Limit length:
 * [x] by number of chars
 * [x] by number of codepoints
 * [x] by number of Unicode symbols (including combining ones)
 * by custom user function (which probably computes the rendered string width in the UI) (postponed: see #20)
* [x] Where to cut the string:
 * [x] between any chars
 * [x] between any codepoints
 * [x] between any Unicode symbols (see `java.text.BreakIterator.getCharacterInstance()`)
 * [x] between any words (see `java.text.BreakIterator.getWordInstance()`)
 * [x] only before delimiter: `"foo, bar..."`
 * [x] only after delimiter: `"foo, bar, ..."`
 * with custom user function (probably using ICU4J `BreakIterator`, etc.) (postponed: see #20)

Roadmap:
* [x] Implementation draft
* [x] Tests
* [x] JavaDoc
* [x] Benchmarks
* [x] Changes
